nano/lib/timestamp.hpp

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <atomic>
+#include <chrono>
+
+namespace nano
+{
+/**
+ * Returns seconds passed since unix epoch (posix time)
+ */
+inline uint64_t seconds_since_epoch ()
+{
+	return std::chrono::duration_cast<std::chrono::seconds> (std::chrono::system_clock::now ().time_since_epoch ()).count ();
+}
+
+/**
+ Creates a unique 64-bit timestamp each time timestamp_now is called.
+ The upper 44-bits are the number of milliseconds since unix epoch
+ The lower 20 bits are a monotonically increasing counter from 0, each millisecond
+ */
+
+class timestamp_generator
+{
+public:
+	static uint64_t component_time (uint64_t value_a)
+	{
+		auto result (value_a & time_mask);
+		return result;
+	}
+	static uint64_t component_count (uint64_t value_a)
+	{
+		auto result (value_a & count_mask);
+		return result;
+	}
+	static uint64_t timestamp_from_ms (uint64_t value_a)
+	{
+		auto result (value_a << count_bits);
+		return result;
+	}
+	static uint64_t now_base ()
+	{
+		uint64_t ms_since_epoch (std::chrono::duration_cast<std::chrono::milliseconds> (std::chrono::system_clock::now ().time_since_epoch ()).count ());
+		uint64_t result (timestamp_from_ms (ms_since_epoch));
+		return result;
+	}
+	uint64_t now ()
+	{
+		static_assert (time_bits + count_bits == 64);
```

Comment:
CI has been updated in `develop` with cpp17 which is needed to support this no-message version of `static_assert`